File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/shader/ShaderBuilder.java
/**
 * 
 */
package com.abubusoft.xenon.shader;

/**
 * @author Francesco Benincasa
 * 
 */
public class ShaderBuilder {

	/**
	 * <p>
	 * </p>
	 * 
	 * @param vertexProgramIdValue
	 * @param fragmentProgramIdValue
	 */
	private ShaderBuilder(int vertexProgramIdValue, int fragmentProgramIdValue, ArgonShaderOptions optionsValue) {
		vertexProgramId = vertexProgramIdValue;
		fragmentProgramId = fragmentProgramIdValue;
		options = optionsValue;
	}
	
	/**
	 * <p>Creatore di builder</p>
	 * 
	 * @param vertexProgramIdValue
	 * @param fragmentProgramIdValue
	 * @param optionsValue
	 * @return
	 * 		builder
	 */
	public static ShaderBuilder build(int vertexProgramIdValue, int fragmentProgramIdValue, ArgonShaderOptions optionsValue)
	{
		ShaderBuilder builder=new ShaderBuilder(vertexProgramIdValue, fragmentProgramIdValue, optionsValue);
		
		return builder;
	}

	/**
	 * <p>
	 * resourceId del vertex program, se esiste. Non viene usata per la
	 * rigenerazione dello shader.
	 * </p>
	 */
	public int vertexProgramId;

	/**
	 * <p>
	 * resourceId del fragment program, se esiste. Non viene usata per la
	 * rigenerazione dello shader.
	 * </p>
	 */
	public int fragmentProgramId;

	/**
	 * <p>
	 * opzioni per lo shader.
	 * </p>
	 */
	public ArgonShaderOptions options;
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/shader/ShaderBuilder.kt
/**
 *
 */
package com.abubusoft.xenon.shader

/**
 * @author Francesco Benincasa
 */
class ShaderBuilder
/**
 *
 *
 *
 *
 * @param vertexProgramIdValue
 * @param fragmentProgramIdValue
 */ private constructor(
    /**
     *
     *
     * resourceId del vertex program, se esiste. Non viene usata per la
     * rigenerazione dello shader.
     *
     */
    var vertexProgramId: Int,
    /**
     *
     *
     * resourceId del fragment program, se esiste. Non viene usata per la
     * rigenerazione dello shader.
     *
     */
    var fragmentProgramId: Int,
    /**
     *
     *
     * opzioni per lo shader.
     *
     */
    var options: ArgonShaderOptions?
) {
    companion object {
        /**
         *
         * Creatore di builder
         *
         * @param vertexProgramIdValue
         * @param fragmentProgramIdValue
         * @param optionsValue
         * @return
         * builder
         */
        @JvmStatic
        fun build(vertexProgramIdValue: Int, fragmentProgramIdValue: Int, optionsValue: ArgonShaderOptions?): ShaderBuilder {
            return ShaderBuilder(vertexProgramIdValue, fragmentProgramIdValue, optionsValue)
        }
    }
}